lib/modules/decoder/StateDecoder.ts

```diff
@@ -0,0 +1,5 @@
+export enum StateDecoder {
```

Comment:
Maybe `DecodingState` is a better name otherwise it's confusing with the Decoder class